### PR TITLE
Update CircleCI config to v2 template

### PR DESCRIPTION
**Pull request by johnrjj**
Hey all,

Upgrading draft-js CircleCI config to v2.
Context: This is the last core repo that's still on CircleCI v1 config. As of the end of this month, v1 configs will unsupported/break on Circle.

There's still some more WIP around deployment, but this gets us caught up to v2.

### CODE REVIEW

File: src/component/handlers/edit/setClipboardData.js [hunk 1]
```diff
@@ -49,8 +49,7 @@ function setClipboardData(e: SyntheticClipboardEvent, editor: DraftEditor, {text
   try {
     clipboard.setData('text/html', html);
     clipboard.setData('text/plain', text);
-  }
-  catch (err) {
+  }  catch (err) {
```
Comment (max-winderbaum) [inline src/component/handlers/edit/setClipboardData.js:52]: Two spaces here?

**Review by max-winderbaum — APPROVED** [review]
:tada:

**Review by david-davidson — APPROVED** [review]
:lgtm:

**Review by fraziermork — APPROVED** [review]
🎉

File: src/component/handlers/edit/setClipboardData.js (hunk 1 above)
Comment (johnrjj) [reply src/component/handlers/edit/setClipboardData.js:52]: Making ESLint happy

**Pull request merged**